src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,115 @@
-            <td className="is-vcentered">7</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">
-                illo expedita consequatur quia in
-              </p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">8</td>
-
-            <td className="is-vcentered">
-              <span className="icon" data-cy="iconCompleted">
-                <i className="fas fa-check" />
-              </span>
-            </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-success">quo adipisci enim quam ut ab</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">9</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">molestiae perspiciatis ipsa</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">10</td>
-
-            <td className="is-vcentered">
-              <span className="icon" data-cy="iconCompleted">
-                <i className="fas fa-check" />
-              </span>
-            </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-success">
-                illo est ratione doloremque quia maiores aut
-              </p>
-            </td>
+export const TodoList: React.FC = () => {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [activeTodoId, setActiveTodoId] = useState<number | null>(null);
+  const todos = useAppSelector((state: RootState) => state.todos.items);
+  const filterState = useAppSelector((state: RootState) => state.filter);
+  const { query, status } = filterState;
+  const dispatch = useDispatch();
+
+  const handleTodoClick = (todo: Todo) => {
+    dispatch(setCurrentTodo(todo));
+    setActiveTodoId(todo.id);
+    setModalOpen(true);
+  };
```

Comment:
The `handleTodoClick` function is created inside the component, which means a new function will be created every time the component re-renders. To avoid unnecessary re-renders of the components where this function is passed as a prop, consider using the `useCallback` hook.